### PR TITLE
Remote MQTT endpoint CA certificate file should be optional

### LINKED ISSUE #34: Remote MQTT endpoint CA certificate file should be optional
Currently, suite-connector cannot start if remote MQTT endpoint CA certificate file is not set. It's not a typical use case but there are some demo setups where plain connection can be used. For example, [Kanto, Explore via Eclipse Hono](https://www.eclipse.org/kanto/docs/getting-started/hono/)

Steps to reproduce:
* remove cacert property from the default configuration file
* restart the suite-connector service

The service start will fail. As it's a configuration validity check, there is no log message. The console dump is:
> 04 10:55:25.166334 Cannot run suite connector: settings validation error: failed to read CA certificate file

### PR DESCRIPTION
**Pull request by antonia-avramova**
[#34 ] Remote MQTT endpoint CA certificate file should be optional

Remove default value of CA certificate file

Signed-off-by: Antonia Avramova <antonia.avramova@bosch.io>

### CODE REVIEW

File: config/connections.go [hunk 1]
```diff
@@ -151,38 +152,46 @@ func CreateHubConnection(
 		honoConfig.BackoffMultiplier = mul
 	}
 
-	tlsConfig, cleaner, err := NewHubTLSConfig(&settings.TLSSettings, logger)
-	if err != nil {
-		return nil, nil, err
+	noCertificates := true
```
Comment (hristobojilov) [inline config/connections.go]: Do  not hard-code logic by using flags.

File: config/connections.go [hunk 2]
```diff
@@ -151,38 +152,46 @@ func CreateHubConnection(
...
+		if !noCertificates {
+			cert = tlsConfig.Certificates[0]
+		}
+
+		honoConfig.TLSConfig = tlsConfig
+	} else {
+		logger.Warnf("Insecure connection is used with address=%s", settings.Address)
 	}
 
-	if len(tlsConfig.Certificates) == 0 {
```
Comment (hristobojilov) [inline config/connections.go:159]: Do not alter the logic this code block

File: config/connections.go [hunk 3]
```diff
@@ -151,38 +152,46 @@ func CreateHubConnection(
 		honoConfig.BackoffMultiplier = mul
 	}
 
-	tlsConfig, cleaner, err := NewHubTLSConfig(&settings.TLSSettings, logger)
-	if err != nil {
-		return nil, nil, err
+	noCertificates := true
+	cert := tls.Certificate{}
```
Comment (hristobojilov) [inline config/connections.go]: It is not possible to use client certificates without TLS enabled.

File: config/connections.go [hunk 4]
```diff
@@ -438,3 +447,12 @@ func HonoConnect(sigs chan os.Signal,
 		}
 	}
 }
+
+func isConnectionSecure(schema string) bool {
+	switch schema {
+	case "wss", "ssl", "tls", "mqtts", "mqtt+ssl", "tcps":
```
Comment (hristobojilov) [inline config/connections.go:447]: Duplicate code

**Review by hristobojilov — CHANGES_REQUESTED** [review]
Please fix these issues

**Review by hristobojilov — APPROVED** [review]

File: config/settings.go [hunk 5]
```diff
@@ -90,8 +91,18 @@ func (settings *Settings) ValidateStatic() error {
...
+	if err != nil {
+		return err
+	}
+	if isConnectionSecure(u.Scheme) {
+		if len(settings.CACert) > 0 {
+			if !util.FileExists(settings.CACert) {
+				return errors.Errorf("failed to read CA certificate file '%s'", settings.CACert)
+			}
+		} else {
+			return errors.Errorf("failed to read CA certificate file '%s'", settings.CACert)
```
Comment (e-grigorov) [inline config/settings.go]: This error message is not correct. That's the case when we have secure endpoint without CA certificates.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**